### ISSUE TITLE
Texts with color mark forced line ending

### ISSUE THREAD

**Issue by DPurple** [themrdemonized/STALKER-Anomaly-modded-exes#6]
It's a bug from original anomaly
While playing game with multibytes texts (eg. chs kor etc...)
Colored tag %c[red green blue etc] will force text block line ending
I took a look at UILines.cpp found it using different behaviors with ANSI version, so i made a patch
But i failed to build a correct binary with those patches
It seem outdated.
There will be a PR referenced to this issues, hope it helps!
Thank you very much.
   P.S  my english sucks

**Comment by themrdemonized** [comment]
Hello, Thanks for your contribution.
I compiled exes just fine with your patch, although i don't know where exactly can i see the changes, i assume in some ui elements like dialogs or descriptions.
What you meant by outdated patches? They are for recent 1.5.2 version of Anomaly. I updated the readme on how to apply them recently cause there was an issue of conflicting patches from shader scopes and screen space shaders.
But anyways, i compiled it fine. I'll merge PR's later if all is fine :)

**Comment by DPurple** [comment]
I applied all patch but weired things happen.
when i enter debug room with F2, i was killed instantly (???)
but with your exes, it work fine
I think that must be something wrong with me if those patches matches last commit
I cloned anomaly 1.5.2 source code from bitbucket
then applied three patches with order : dltx-bugfix.diff shader-scopes-v2.diff sss.diff
some whitespace warning appeared. but no errors generated
the last one sss.diff get some shift and failed to apply, i merged them by hand
after all, I convert all source code from cp1251 into UTF-8 (dealed with some file with UTF-8 DOM head, and some comment failed to convert) with iconv
finally everything compiles and all exes generated.
Then i made a commit on local git repository and modified that UILines.cpp.patch

Sorry I thought it was outdated because some commit of binaries didn't matches patch file
it confused me.. I can post some pictures for this bug, it's has been years, but it's not critical
